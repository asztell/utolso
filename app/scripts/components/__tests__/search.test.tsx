import React from "react";
import { IntlProvider } from "react-intl";
import { render, screen, fireEvent } from "@testing-library/react";
import { ProductsProvider } from "../../contexts/products";
import { Search } from "../search";
import { fetchProductsByName } from "../../utils/services";
import { act } from "react-dom/test-utils";

jest.mock("../../utils/services", () => ({
  fetchProductsByName: jest.fn(() => Promise.resolve([])),
}));

describe("<Search />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders", () => {
    render(
      <IntlProvider
        locale={navigator.language}
        messages={{
          "SearchInput.Placeholder": "test SEARCH",
        }}
      >
        <ProductsProvider>
          <Search />
        </ProductsProvider>
      </IntlProvider>
    );
    expect(screen.getByPlaceholderText("test SEARCH")).toBeInTheDocument();
    act(() => {
      fireEvent.change(screen.getByPlaceholderText("test SEARCH"), {
        target: { value: "test" },
      });
    });
    expect(screen.getByPlaceholderText("test SEARCH")).toHaveValue("test");
  });
});
