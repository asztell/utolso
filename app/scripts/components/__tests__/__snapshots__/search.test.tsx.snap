// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders 1`] = `
<body>
  <div>
    <div
      class="Search box"
    >
      <input
        placeholder="test SEARCH"
        type="text"
      />
    </div>
  </div>
</body>
`;
